ui/src/app/shared/utils/resource-utils.ts

```diff
@@ -3,19 +3,34 @@ export class ResourceUtils {
   static browserPrefix: string = "https://console.cloud.google.com/storage/browser/";
   static storagePrefix: string = "https://storage.cloud.google.com/";
 
+  /** Return boolean indicating if the value is a GCS url to a resource */
+  public static isResourceURL(value: string): boolean {
+    return !!ResourceUtils.validateGcsURLGetParts(value);
+  }
+
   /** Get link to a file's enclosing directory from its gcs url */
-  public static getResourceBrowserURL(uri: string): string {
-    let parts = ResourceUtils.validateGcsURLGetParts(uri);
+  public static getResourceBrowserURL(url: string): string {
+    let parts = ResourceUtils.validateGcsURLGetParts(url);
     // This excludes the object from the link to show the enclosing directory.
     // This is valid with wildcard glob (bucket/path/*) and directories
     // (bucket/path/dir/) as well, the * or empty string will be trimmed.
-    return parts ? ResourceUtils.browserPrefix + parts.slice(2,-1).join("/") : undefined;
+    if (!parts) {
+      return undefined;
+    }
+
+    var browserUrl = ResourceUtils.browserPrefix + parts.slice(2,-1).join("/");
+    if (url.indexOf('*') == -1) {
+      browserUrl += "?prefix=" + ResourceUtils.getResourceFileName(url);
```

Comment:
Cool

Reply:
Yeah I forgot to mention this addition in the description. Its actually quite nice when there's a bunch a junk all in the same directory this URL will filter it. Unfortunately doesn't work with wildcards though.